packages/app-desktop/gui/NoteEditor/utils/useMessageHandler.ts

```diff
@@ -85,8 +85,8 @@ export default function useMessageHandler(scrollWhenReady: any, setScrollWhenRea
 			}
 		} else if (urlUtils.urlProtocol(msg)) {
 			if (msg.indexOf('file://') === 0) {
-				// When using the file:// protocol, openExternal doesn't work (does nothing) with URL-encoded paths
-				require('electron').shell.openExternal(urlDecode(msg));
+				// When using the file:// protocol, openPath doesn't work (does nothing) with URL-encoded paths
+				require('electron').shell.openPath(urlDecode(msg));
```

Comment:
How do we know for sure this change didn't break something else? I seem to remember it was openPath before, then we switch to openExternal, and now back to openPath. Instead we should make sure we fully understand the problem. Why is it not possible to get it working with openExternal for example?

Reply:
> How do we know for sure this change didn't break something else?

I am quite confident because here `openPath` is used only if the link is actually a file path.  And `openPath`  efficiently handles file paths/URLs ( as it is built for opening file paths ) containing Unicode characters. 

> I seem to remember it was openPath before, then we switch to openExternal, and now back to openPath. Instead we should make sure we fully understand the problem. 

Actually, I have tested with various files (pdfs, txt files etc.) as this code change can affect at-most only the file paths ( and nothing else ) and all the tests successfully passed.

> Why is it not possible to get it working with openExternal for example?

I strongly believe that it is an `electron` issue, also it is possible that this issue exists only on windows.

It seems that `shell.openExternal` does not work correctly with file paths containing Unicode characters. Though they made a fix for this (see https://github.com/electron/electron/issues/6302 ), it only works for URLs and not File paths ( at least on windows ).

Reply:
Again, if you check the electron code, `openPath` expects a file path of type `base::FilePath&` whereas `openExternal` expects a URL of type `GURL&`.

https://github.com/electron/electron/blob/4da01641ce50cf5a8b6f39717e4c447f89455dfa/shell/common/platform_util_win.cc#L243
Here, in the `OpenExternalOnWorkerThread` function, it returns `"Failed to open"` when the condition in the if statement is true and exactly the same message gets displayed when we try to open a file link with German Umlauts by clicking on it.

It looks like( I am not sure ), the problem is with the chromium's `url.spec()` ( see https://chromium.googlesource.com/chromium/src.git/+/refs/heads/main/url/gurl.h#104 ).

On the other hand, openPath directly passes the given file path to the `ui::win::OpenFileViaShell(full_path)` function.

Reply:
Thanks for clarifying, that makes a lot of sense. I'll test on macOS too just to be sure and then we can merge.